src/doc/guide.md

```diff
@@ -455,38 +449,41 @@ x = 10i;
 There is no single reason that bindings are immutable by default, but we can
 think about it through one of Rust's primary focuses: safety. If you forget to
 say `mut`, the compiler will catch it, and let you know that you have mutated
-something you may not have cared to mutate. If bindings were mutable by
-default, the compiler would not be able to tell you this. If you _did_ intend
-mutation, then the solution is quite easy: add `mut`.
+something you may not have cared to mutate. If bindings were mutable by default,
+the compiler would not be able to tell you this. If you _did_ intend mutation,
+then the solution is quite easy: add `mut`.
 
 There are other good reasons to avoid mutable state when possible, but they're
 out of the scope of this guide. In general, you can often avoid explicit
-mutation, and so it is preferable in Rust. That said, sometimes, mutation is
-what you need, so it's not verboten.
+mutation, so it is preferable in Rust. That said, sometimes mutation is what you
+need, so it's not verboten.
 
-Let's get back to bindings. Rust variable bindings have one more aspect that
-differs from other languages: bindings are required to be initialized with a
-value before you're allowed to use them. If we try...
+Let's get back to bindings. Rust `let` bindings have one more aspect that
+differs from other languages: they cannot be left uninitialized:
 
 ```{ignore}
 let x;
 ```
 
-...we'll get an error:
+We'll get an error:
 
 ```text
-src/main.rs:2:9: 2:10 error: cannot determine a type for this local variable: unconstrained type
+src/main.rs:2:9: 2:10 error: unable to infer enough type information to locate the impl of the trait `core::kinds::Sized` for the type `<generic #1>`; type annotations required
+src/main.rs:2     let x;
+                      ^
+src/main.rs:2:9: 2:10 note: all local variables must have a statically known size
```

Comment:
I think so. I got it from somewhere! play.rust-lang.org shows that master has roughly the original error message but that 0.12.0 has roughly the new error message. Maybe I should revert? :/

:+1: